src/commands/add.ts

```diff
@@ -0,0 +1,54 @@
+import { getConfig } from "../utils/getConfig.js";
+import ora from "ora";
+import { select } from "../utils/questions/select.js";
+import { createGitHubCredential } from "../utils/createGitHubCredential.js";
+import { addCredential } from "../utils/addCredential.js";
+import { makeActive } from "../utils/makeActive.js";
+
+export const add = async () => {
+  const spinner = ora("Add credential process started.").start();
+
+  const { credentialsPath, credentialsExists } = await getConfig();
+
+  let projectManagementSystem;
+  let newCredential;
+
+  if (!credentialsExists) {
+    spinner.fail('You must run "ttb init" before you can add a credential.');
+    return;
+  }
+
+  spinner.stop();
+
+  projectManagementSystem = await select(
+    "What project management system are you using?",
+    ["GitHub", "Jira", "Trello"]
+  );
+
+  if (projectManagementSystem && projectManagementSystem === "GitHub") {
+    newCredential = await createGitHubCredential();
+  }
+
+  if (projectManagementSystem && projectManagementSystem === "Jira") {
+    spinner.fail("Jira is not yet supported");
+    return;
+  }
+
+  if (projectManagementSystem && projectManagementSystem === "Trello") {
+    spinner.fail("Trello is not yet supported");
+    return;
+  }
+
+  spinner.start("Editing config file");
+
+  await addCredential(newCredential);
```

Comment:
It smells weird to have a side effect like this one without a return value. The return value is a very healthy thing where we can determine in the upper scope if we are still on happy-path or not.

Reply:
Fixed